src/sonic-containercfgd/containercfgd/containercfgd.py

```diff
@@ -0,0 +1,198 @@
+import subprocess
+import sys
+
+from sonic_py_common import daemon_base, logger
+from swsscommon.swsscommon import ConfigDBConnector, RestartWaiter
+
+SYSLOG_IDENTIFIER = "containercfgd"
+logger = logger.Logger(SYSLOG_IDENTIFIER)
+
+# Table names
+FEATURE_TABLE = 'FEATURE'
+SYSLOG_CONFIG_FEATURE_TABLE = 'SYSLOG_CONFIG_FEATURE'
+
+# Table field names
+SYSLOG_RATE_LIMIT_INTERVAL = 'rate_limit_interval'
+SYSLOG_RATE_LIMIT_BURST = 'rate_limit_burst'
+
+# Container name
+container_name = None
+
+
+def run_command(command):
+    """
+    Utility function to run an shell command and return the output.
+    :param command: Shell command string.
+    :return: Output of the shell command.
+    """
+    return subprocess.check_output(shlex.split(command), text=True, stderr=subprocess.PIPE)
+
+
+class ContainerConfigDaemon(daemon_base.DaemonBase):
+    handlers = {}
+
+    def __init__(self):
+        super(ContainerConfigDaemon, self).__init__(SYSLOG_IDENTIFIER)
+
+    def run(self):
+        """Register config handlers and listen to CONFIG DB changes
+        """
+        config_db = ConfigDBConnector()
+        config_db.connect(wait_for_init=True, retry_on=True)
+        self.log_notice(f'Connected to CONFIG DB')
+        for table_name, handler in self.handlers.items():
+            config_db.subscribe(table_name, handler.handle_config)
+        config_db.listen(init_data_handler=self.init_data_handler)
+
+    def init_data_handler(self, init_data):
+        """Handle initial data in CONFIG DB
+
+        Args:
+            init_data (dict): Initial data when first time connecting to CONFIG DB. {<table_name>: {<field_name>: <field_value>}}
+        """
+        for handler in self.handlers.values():
+            handler.handle_init_data(init_data)
+
+    @classmethod
+    def register_handler(cls, table_name, object_type):
+        """Register CONFIG DB handler
+
+        Args:
+            table_name (str): CONFIG DB table name
+            object_type (class): Class of CONFIG DB handler
+        """
+        cls.handlers[table_name] = object_type()
+
+    def signal_handler(self, sig, frame):
+        if sig == signal.SIGHUP:
+            self.log_info("ContainerCfgd: Caught SIGHUP - ignoring...")
+        elif sig == signal.SIGINT:
+            self.log_info("ContainerCfgd: Caught SIGINT - exiting...")
+            sys.exit(128 + sig)
+        elif sig == signal.SIGTERM:
+            self.log_info("ContainerCfgd: Caught SIGTERM - exiting...")
+            sys.exit(128 + sig)
+        else:
+            self.log_warning("ContainerCfgd: Caught unhandled signal '{}'".format(sig))
+
+
+def config_handler(table_name):
+    """Decorator to register CONFIG DB handler
+
+    Args:
+        table_name (str): CONFIG DB table name
+    """
+    def wrapper(object_type):
+        ContainerConfigDaemon.register_handler(table_name, object_type)
+        return object_type
+    return wrapper
+
+
+@config_handler(SYSLOG_CONFIG_FEATURE_TABLE)
+class SyslogHandler:
```

Comment:
This seems very similar to the code in the sonic-host-services pull request.  Is there any way to share that?

Reply:
This is good suggestion. I also thought of sharing the logic between sonic-host-services and containercfgd, however, I finnaly decided to leave it as is because:

1. cannot find a good place to do so. The logic for SyslogHandler is not so general to put it into sonic_py_common. 
2. The logic in sonic-host-services and containercfgd are similar but not exactly the same.